server/tests/batch_comparison/test_multi_contest_batch_comparison.py

```diff
@@ -0,0 +1,1003 @@
+from typing import Dict, List
+import pytest
+from flask.testing import FlaskClient
+
+from ...models import *  # pylint: disable=wildcard-import
+from ..helpers import *  # pylint: disable=wildcard-import
+
+
+@pytest.fixture
+def contest_ids(client: FlaskClient, election_id: str, jurisdiction_ids: List[str]):
+    set_logged_in_user(client, UserType.AUDIT_ADMIN, DEFAULT_AA_EMAIL)
+    contests = [
+        {
+            "id": str(uuid.uuid4()),
+            "name": "Contest 1",
+            "isTargeted": True,
+            "choices": [
+                {"id": str(uuid.uuid4()), "name": "Candidate 1", "numVotes": 750},
+                {"id": str(uuid.uuid4()), "name": "Candidate 2", "numVotes": 250},
+            ],
+            "numWinners": 1,
+            "votesAllowed": 1,
+            "jurisdictionIds": jurisdiction_ids,
+        },
+        {
+            "id": str(uuid.uuid4()),
+            "name": "Contest 2",
+            "isTargeted": True,
+            "choices": [
+                {"id": str(uuid.uuid4()), "name": "Candidate 3", "numVotes": 450},
+                {"id": str(uuid.uuid4()), "name": "Candidate 4", "numVotes": 50},
+            ],
+            "numWinners": 1,
+            "votesAllowed": 2,
+            "jurisdictionIds": [jurisdiction_ids[0]],
+        },
+    ]
+    rv = put_json(client, f"/api/election/{election_id}/contest", contests)
+    assert_ok(rv)
+    return [str(c["id"]) for c in contests]
+
+
+@pytest.fixture
+def manifests(client: FlaskClient, election_id: str, jurisdiction_ids: List[str]):
+    set_logged_in_user(client, UserType.AUDIT_ADMIN, DEFAULT_AA_EMAIL)
+    manifests_by_jurisdiction = {
+        jurisdiction_ids[0]: io.BytesIO(
+            b"Batch Name,Number of Ballots\n"
+            b"Batch 1,100\n"
+            b"Batch 2,100\n"
+            b"Batch 3,100\n"
+            b"Batch 4,100\n"
+            b"Batch 5,100\n"
+            b"Batch 6,100\n"
+            b"Batch 7,100\n"
+            b"Batch 8,100\n"
+            b"Batch 9,100\n"
+            b"Batch 10,100\n"
+        ),
+        jurisdiction_ids[1]: io.BytesIO(
+            b"Batch Name,Number of Ballots\n"
+            b"Batch 1,100\n"
+            b"Batch 2,50\n"
+            b"Batch 3,50\n"
+            b"Batch 4,50\n"
+        ),
+        jurisdiction_ids[2]: io.BytesIO(
+            b"Batch Name,Number of Ballots\n"
+            b"Batch 1,100\n"
+            b"Batch 2,50\n"
+            b"Batch 3,50\n"
+            b"Batch 4,50\n"
+        ),
+    }
+    for jurisdiction_id, manifest in manifests_by_jurisdiction.items():
+        rv = client.put(
+            f"/api/election/{election_id}/jurisdiction/{jurisdiction_id}/ballot-manifest",
+            data={"manifest": (manifest, "manifest.csv",)},
+        )
+        assert_ok(rv)
+
+
+VALID_BATCH_TALLIES = [
```

Comment:
What happens with the batch tallies file if a candidate runs in multiple contests? Rare, but I've seen it

Reply:
Oh great to know that this is possible! I'd forgotten that I'd made this assumption.

I'm thinking of disambiguating with the contest name but only when necessary, e.g.,

```
Batch Name,Choice 1,Choice 2 (Contest 1),Choice 3,Choice 2 (Contest 2)
```

Once I finish https://github.com/votingworks/arlo/issues/1856 (auto-generating candidate-totals-by-batch templates), the exact format won't matter too much

Reply:
If we're auto-generating a template for the file, I wonder if it would be clearer to always include the contest name when there are multiple contests

Another option for format is Dominion CVR style: two stacked header rows, one with contest names, one with candidate names:
```
          ,Contest 1,Contest 1,Contest 2
Batch Name,Choice 1,Choice 2,Choice 1
````
Not sure I have a preference

Reply:
> I wonder if it would be clearer to always include the contest name when there are multiple contests

Agreed! Went with that: https://github.com/votingworks/arlo/pull/1884/commits/7af04b0a2e56fc0e419ef763dbd1285d6c0767ce